src/extension.ts

```diff
@@ -507,9 +507,11 @@ function registerCommands(
           let { activeFile, allFiles } = activeFiles();
 
           let request: ExecuteCommandParams = {
-            command: "sourcery/chat/request",
+            command: "sourcery.coding_assistant",
             arguments: [
               {
+                view: "chat",
+                request: "sendMessage",
                 message: message,
                 selected: selectionLocation,
                 active_file: activeFile,
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏I'd be tempted to group message, selected, and active_file into a `body`

Reply:
Flat interfaces are easier to create types for and query (via dictionary lookups, for example). Can you explain why you'd add a layer of nesting here?

Reply:
So that different webapp requests could have different shapes. e.g. a chat request may have a `{message: ...}` body while a code review request may have a `{from_branch: ..., to_branch: ...)` shape. Then in the backend they can be deserialized into classes

Reply:
You don't need a body to do that - it's just different union types, disambiguated by the view/request keys. In your example, we'd just have for the chat:
```json
{
  view: "chat",
  request: "sendMessage",
  message: "Example"
}
```
and for the review:
```
{
  view: "review",
  request: "sendMessage",
  fromBranch: "main",
  toBranch: "example"
}
```

Reply:
(I fully agree with having different shapes for different requests, but you don't need nesting to achieve that)

Reply:
That would work fine, I'm just wondering if it provides the nicest webapp and backend interfaces.

e.g. It would be nice to be able to write something like this:
```js
window.sourceryLS.sendRequest("codeReview", {from_branch: ..., "to_branch"})
```

and use it on the backend:
```python
class CodeReviewRequest:
    from_branch: str
    to_branch: str
    
def handle_code_review_request(self, request: CodeReviewRequest): ...
```

Essentially I'm suggesting to separate routing from payload

However if you don't think it's right then I don't mind

Reply:
Yeah that's fair - I do get the principle of what you're saying but in practice I tend to find it's easier to refactor and work with messages which have this flat structure. It means that if we change fields or message types in future, it's much easier to distribute those among various handlers that need them, and all the information about a request is included in one place. It also makes it harder to accidentally send the wrong payload with a particular request. I'm going to leave for now, but we'll keep an eye on the system and if it's proving difficult to work with I'll reconsider.